### PR TITLE
[WIP] Asset-buyer 2.0 Exchange SwapQuote Consumer

### PR DESCRIPTION
**Pull request by dave4506**
## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/asset-buyer/src/quote_consumers/exchange_swap_quote_consumer.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        const calldataHexString = abiEncoder.encode(args);
+        return {
+            calldataHexString,
+            methodAbi,
+            to,
+            ethAmount,
+        };
+    }
+
+    public async getSmartContractParamsOrThrowAsync(quote: SwapQuote, opts: Partial<SwapQuoteGetOutputOpts>): Promise<SmartContractParamsInfo<ExchangeMarketBuySmartContractParams>> {
```
Comment (fragosti) [inline packages/asset-buyer/src/quote_consumers/exchange_swap_quote_consumer.ts]: I don't think you're using the `opts` here.

File: packages/asset-buyer/src/quote_consumers/exchange_swap_quote_consumer.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            params,
+            to: this._contractWrappers.exchange.address,
+            methodAbi,
+        };
+    }
+
+    public async executeSwapQuoteOrThrowAsync(quote: SwapQuote, opts: Partial<SwapQuoteExecutionOpts>): Promise<string> {
+        assert.isValidSwapQuote('quote', quote);
+
+        const { takerAddress, gasLimit, gasPrice } = opts;
```
Comment (fragosti) [inline packages/asset-buyer/src/quote_consumers/exchange_swap_quote_consumer.ts]: No default values here?

Comment (dave4506) [reply packages/asset-buyer/src/quote_consumers/exchange_swap_quote_consumer.ts]: These values are all optional so there is no need for a default value obj.

File: packages/asset-buyer/src/quote_consumers/exchange_swap_quote_consumer.ts (hunk 1 above)
Comment (dave4506) [reply packages/asset-buyer/src/quote_consumers/exchange_swap_quote_consumer.ts]: Later, when supporting marketSell, the opts will probably be used.

**Pull request merged**